components/Spaces/CallTabs/Music/Tabs/Queue.js

```diff
@@ -13,10 +13,32 @@ const useStyles = makeStyles((theme) => ({
 
 const Queue = () => {
   const classes = useStyles();
-  const { currentTrack, nextTracks, setTrackUri } = useSpotify();
+  const { queue, setQueue, currentTrack, nextTracks, setNextTracks } = useSpotify();
 
   const playTrack = (track) => {
-    setTrackUri(track?.uri);
+    let found = false;
+    const queueClone = [...queue];
+    const newQueue = queueClone.reduce((acc, t) => {
+      if (found) return acc.concat([t]);
+      if (t.title === track.title && t.artist === track.artist) {
+        console.debug(t);
+        found = true;
+        return [t];
+      }
+      return [];
+    }, []);
+    setQueue(newQueue);
```

Comment:
maybe comments here.
what is `found` for? Its initially `false` so the first couple songs dont get added until we find a specific track? can we do the same with pop?

Reply:
I'll add documentation.